pkg/vars/manager.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vars
+
+import (
+	"sort"
+
+	"github.com/a8m/envsubst/parse"
+)
+
+// GlobalVarManager is the global instance of the Okteto Variables manager. It should only be used in the serializer where it's harder to inject the manager.
+var GlobalVarManager *Manager
```

Comment:
> We are using it outside the serialiser for adding variables?

No, there was only one instance which I've removed now: a5c20db815cb6a54978b1239c5d4d76afb5038c9. All other usages are and should be the serializers.

> Should we use it just on the serialised methods? How much effort do you think we should add in order to get rid of this?

Once we touch all serializers, for example when we decide to parse the manifest in sections instead as a whole file, it could be something the tackle then, otherwise I would not invest in it.